aten/src/ATen/native/mps/operations/FusedAdamAmsgradKernelImpl.h

```diff
@@ -20,5 +20,23 @@ void _fused_adam_amsgrad_mps_impl_(
     const c10::optional<at::Tensor>& grad_scale,
     const c10::optional<at::Tensor>& found_inf
 );
+
+void _fused_adam_amsgrad_mps_impl_(
+    at::TensorList params,
+    at::TensorList grads,
+    at::TensorList exp_avgs,
+    at::TensorList exp_avg_sqs,
+    at::TensorList max_exp_avg_sqs,
+    at::TensorList state_steps,
+    const at::Tensor& lr,
+    const double beta1,
+    const double beta2,
+    const double weight_decay,
+    const double eps,
+    const bool maximize,
+    const c10::optional<at::Tensor>& grad_scale,
+    const c10::optional<at::Tensor>& found_inf
+);
```

Comment:
Q: Why does it need to be declared there? I.e. do you expect this function to be called outside of FusedAdamAmsgradKernelImpl.mm? If not, than just declare it static there

Reply:
It is called here:

https://github.com/pytorch/pytorch/blob/e1ec81f9134567dfa80d230538bc616ea6f26a37/aten/src/ATen/native/mps/operations/FusedAdamKernel.mm#L109-L122

Reply:
Can you please elaborate on this `XYZ.mm` and `XYZImpl.mm` pattern? Why split the two and introduce header files (and function signatures) which seems larger than the actual function implementaiton?

Reply:
The reason was that I used the CUDA implementation as the template when I was implementing it, and it is how the CUDA implementation is written. I agree with you that these are kind of verbose and we can definitely do BE on them.

Reply:
Please submit a followup BE change then.